chore: only one fork when block building

just-mitch opened the pull request:
Now sequencers and validators only use a single world state fork while building blocks, and take advantage of the fact that the public processor will do tree insertions.

Note: this does not modify provers, who still do double insertions for now.

Fix #11000

just-mitch commented:
* **#14475** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/14475?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/14475?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

alexghr reviewed (approved):
Looks great!

PhilWindle reviewed (approved):
Great stuff. A relatively simple change too. Thanks

critesjosh committed:
chore(docs): Update alpha-testnet docs to use version 0.87.2 (#14501)

Updates docs to use the appropriate version tag. Also updates the
protocol contracts on the "Try Testnet" page.

Removes v0.86.0 docs
